bundles/org.openhab.binding.gridbox/src/main/java/org/openhab/binding/gridbox/internal/api/GridBoxApi.java

```diff
@@ -0,0 +1,277 @@
+            logger.trace("Response body: {}", body);
+            throw new GridBoxApiException("Gateway request returned an invalid response");
+        }
+
+        logger.atTrace().log(() -> "Gateway request returned body: {}".formatted(body));
+
+        Optional<String> systemIdValue = parseSystemIdValue(body);
+        if (systemIdValue.isEmpty()) {
+            logger.debug("Invalid response of gateway request, returned status {}", status);
+            logger.trace("Response body: {}", body);
+            throw new GridBoxApiException("Gateway request returned an invalid response");
+        }
+
+        return systemIdValue.get();
+    }
+
+    /**
+     * Parse the System ID from a JSON response body
+     *
+     * @param body A JSON response containing the System ID in an "id" attribute
+     * @return The parsed "id" value, only non-empty if parsing was successful
+     */
+    public static Optional<String> parseSystemIdValue(String body) {
+        JsonElement jsonObject = JsonParser.parseString(body);
+        String systemIdValue = null;
+        if (jsonObject.isJsonArray()) {
+            JsonElement index0Element = jsonObject.getAsJsonArray().get(0);
+            if (index0Element != null && index0Element.isJsonObject()) {
+                JsonElement systemElement = index0Element.getAsJsonObject().get("system");
+                if (systemElement != null && systemElement.isJsonObject()) {
+                    JsonElement idElement = systemElement.getAsJsonObject().get("id");
+                    if (idElement != null && idElement.isJsonPrimitive()) {
+                        systemIdValue = idElement.getAsString();
+                    }
+                }
+            }
+        }
+        return Optional.ofNullable(systemIdValue);
+    }
+
+    /**
+     * Fetch the Live Data from the GridBox API by executing an API query.
+     *
+     * @param config The {@link GridBoxConfiguration} to use, containing a valid ID token
+     * @param responseHandler A function handling the retrieved live data, called if a valid response was received
+     *            with a {@link LiveData} object containing the content of the response.
+     * @throws IOException see {@link HttpClient#send(HttpRequest, java.net.http.HttpResponse.BodyHandler)}
+     * @throws InterruptedException see {@link HttpClient#send(HttpRequest, java.net.http.HttpResponse.BodyHandler)}
+     * @throws GridBoxApiException Thrown in case of an invalid response of the live data query, either
+     *             if a status code not equal to 200 or 403 is returned or the {@link LiveData} instance cannot be
+     *             parsed from the response body
+     * @throws GridBoxApiSystemNotFoundException Thrown in case the query returned status code 404, representing a
+     *             non-existing System ID
+     * @throws GridBoxApiAutheticationException Thrown in case of an invalid response of the authorization query if a
```

Comment:
```suggestion
     * @throws GridBoxApiAuthenticationException Thrown in case of an invalid response of the authorization query if a
```

Please check all occurrences.